src/Stryker.Core/Stryker.Core/ToolHelpers/VsTestHelper.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Stryker.Core.Options;
+using Stryker.Core.Options.Values;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Stryker.Core.ExecutableFinders
+{
+    public class VsTestHelper
+    {
+        private readonly StrykerOptions _options;
+        private readonly IFileSystem _fileSystem;
+
+        public VsTestHelper(StrykerOptions options, IFileSystem fileSystem = null)
+        {
+            _options = options;
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public Dictionary<DotnetFramework, string> GetVsTestToolPaths()
+        {
+            Dictionary<DotnetFramework, string> vsTestPaths = new Dictionary<DotnetFramework, string>();
+            string versionString = "16.0.0-preview-20181205-02";
+            string portablePackageName = "microsoft.testplatform.portable";
+
+            var nugetPackageFolders = CollectNugetPackageFolders();
+
+            bool dllFound = false;
+            bool exeFound = false;
+            foreach (string nugetPackageFolder in nugetPackageFolders)
+            {
+                if (dllFound && exeFound)
+                {
+                    break;
+                }
+
+                string portablePackageFolder = _fileSystem.Directory.GetDirectories(nugetPackageFolder, portablePackageName, SearchOption.AllDirectories).First();
+
+                string dllPath = FilePathUtils.ConvertPathSeparators(
```

Comment:
Can't we search the whole `portablePackageFolder` for files with the name: `vstest.console` as a filter. That way even if the path insided the package ever changes we can still locate vstest

Reply:
We need those specific versions of vstest.console. If the location changes we will have to investigate whether we are still compatible anyhow and our integration tests will fail on it so I do not see any problem in keeping it how it is.